apps/builder/app/canvas/canvas.tsx

```diff
@@ -130,21 +131,46 @@ const useElementsTree = (
 };
 
 const DesignMode = () => {
-  useManageDesignModeStyles();
+  const debounceEffect = useDebounceEffect();
+  const ref = useRef<Instances>();
+
   useDragAndDrop();
-  // We need to initialize this in both canvas and builder,
-  // because the events will fire in either one, depending on where the focus is
-  // @todo we need to forward the events from canvas to builder and avoid importing this
-  // in both places
-  useCopyPaste();
 
-  useScrollNewInstanceIntoView();
-  useSelectedInstance();
-  useEffect(updateCollaborativeInstanceRect, []);
-  useEffect(subscribeInstanceSelection, []);
-  useEffect(subscribeInstanceHovering, []);
-  useEffect(subscribeInspectorEdits, []);
-  useEffect(subscribeFontLoadingDone, []);
+  useEffect(() => {
+    const abortController = new AbortController();
+    subscribeScrollNewInstanceIntoView(
+      debounceEffect,
+      ref,
+      abortController.signal
+    );
+    const unsubscribeSelected = subscribeSelected(debounceEffect);
+    return () => {
+      unsubscribeSelected();
+      abortController.abort();
+    };
+  }, [debounceEffect]);
+
+  useEffect(() => {
+    return subscribeSelected(debounceEffect);
+  }, [debounceEffect]);
+
+  useEffect(() => {
+    const abortController = new AbortController();
+    // We need to initialize this in both canvas and builder,
+    // because the events will fire in either one, depending on where the focus is
+    // @todo we need to forward the events from canvas to builder and avoid importing this
+    // in both places
+    initCopyPaste(abortController);
```

Comment:
why not pass signal explicitly?

Reply:
idk, thought its unnecessary typing ....